tests/functional/prometheous/test_prometheus_validations.py

```diff
@@ -974,3 +975,81 @@ def test_restart_prometheus_glusterfs_pod(self):
             # Try to fetch metric from prometheus pod
             self._fetch_metric_from_promtheus_pod(
                 metric='heketi_device_brick_count')
+
+    @pytest.mark.tier2
+    def test_metrics_workload_on_prometheus(self):
+        """Validate metrics workload on prometheus"""
+
+        # Skip test if the prometheus pods are not present
+        openshift_ops.switch_oc_project(
+            self._master, self._prometheus_project_name)
+        prometheus_pods = openshift_ops.oc_get_pods(
+            self._master, selector=self._prometheus_resources_selector)
+        if not prometheus_pods:
+            self.skipTest(
+                prometheus_pods, "Skipping test as prometheus"
+                " pod is not present")
+
+        self.prefix = "autotest-{}".format(utils.get_random_str())
+        if not self.registry_sc:
+            self.skipTest(
+                prometheus_pods, "Skipping test as registry "
+                " storage details are not provided")
+        self._registry_project = self.registry_sc.get(
+            'restsecretnamespace')
+
+        # Get one of the prometheus pod name and respective pvc name
+        prometheus_pod = list(prometheus_pods.keys())[0]
+        pvc_custom = ":.spec.volumes[*].persistentVolumeClaim.claimName"
+        pvc_name = openshift_ops.oc_get_custom_resource(
+            self._master, "pod", pvc_custom, prometheus_pod)
+        self.assertTrue(
+            pvc_name, "Failed to get PVC name for prometheus"
+            " pod {}".format(prometheus_pod))
+        pvc_name = pvc_name[0]
+        self.verify_iscsi_sessions_and_multipath(
+            pvc_name, prometheus_pod, rtype='pod',
+            heketi_server_url=self._registry_heketi_server_url,
+            is_registry_gluster=True)
+
+        # Try to fetch metric from the prometheus pod
+        self._fetch_metric_from_promtheus_pod(
+            metric='kube_persistentvolumeclaim_info')
+
+        # Create storage class
+        openshift_ops.switch_oc_project(
+            self._master, self._registry_project)
+        self.sc_name = self.create_storage_class(
+            vol_name_prefix=self.prefix, glusterfs_registry=True)
+        self.addCleanup(openshift_ops.switch_oc_project,
```

Comment:
Why switching back to `self._registry_project` as you are switching to `self._registry_project`  in line number 1020, So the clean will be to the project from where it was previously i.e before switching to `self._registry_project`

Reply:
You are suggesting we do not need the cleanup here for switching to self._registry_project ?

Reply:
No, I mean add a check to know in which project currently you are, i.e before switching and then in clean up use the project to switch back

Reply:
@SushilG96 in this step storage class is created in `self._registry_project` and cleanup to delete same sc is added in the function without mention of namespace. But in this TC at few steps namespace is changed to fetch metrics from prometheus, which will cause issue while executing the cleanup to delete sc created in `self._registry_project`. To avoid this situation we are adding cleanup to move `self._registry_project`  namespace.

Reply:
Using 'self._registry_project' in addCleanup() at Line 1024.